app/src/main/java/com/nextlevelstudy/ui/login/LoginFragment.java

```diff
@@ -0,0 +1,141 @@
+import static com.nextlevelstudy.services.utils.Status.SUCCESS;
+
+
+public class LoginFragment extends Fragment implements Injectable {
+
+  @Inject
+  public ViewModelProvider.Factory viewModelFactory;
+  @Inject
+  public NavigationController navigationController;
+
+  @Inject
+  public AuthInterceptor tokenAuthInterceptor;
+
+  @BindView(R.id.login_email)
+  EditText loginEmail;
+  @BindView(R.id.login_password)
+  EditText loginPassword;
+  @BindView(R.id.login_authenticate)
+  Button authenticateButton;
+  @BindView(R.id.login_progressbar)
+  ProgressBar progressBar;
+  @BindView(R.id.login_signUp_text)
+  TextView signUp;
+  @BindView(R.id.error_msg)
+  TextView errorTextView;
+  @BindView(R.id.layout_display_info)
+  RelativeLayout layoutDisplayInfo;
+
+  // viewmodel
+  LoginViewModel loginViewModel;
+  // adapter
+  LoginAdapter loginAdapter;
+
+
+  @Nullable
+  @Override
+  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+                           @Nullable Bundle savedInstanceState) {
+
+    View view = inflater.inflate(R.layout.fragment_login, container, false);
+    ButterKnife.bind(this, view);
+
+    authenticateButton.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View v) {
+        loginViewModel.authenticate(
+          loginEmail.getText().toString(),
+          loginPassword.getText().toString());
+      }
+    });
+
+    return view;
+  }
+
+  @Override
+  public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+    super.onActivityCreated(savedInstanceState);
+
+    DeviceUtils.setTranslucentStatusBar(getActivity().getWindow(), R.color.colorPrimaryDark);
+
+    // get the view model
+    loginViewModel = ViewModelProviders.of(this, viewModelFactory)
+      .get(LoginViewModel.class);
+
+    loginViewModel.getStudent().observe(this, this::handleLoginResponse);
+
+    signUp.setOnClickListener(v -> {
+      navigationController.navigateToStudentRegistration();
+    });
+  }
+
+  private void handleLoginResponse(Resource<StudentWithToken> loginResultResource) {
+    if (loginResultResource != null) {
+      switch (loginResultResource.status) {
+        case LOADING:
+          layoutDisplayInfo.setVisibility(View.VISIBLE);
+          progressBar.setVisibility(View.VISIBLE);
+          errorTextView.setVisibility(View.GONE);
+          break;
+        case ERROR:
+          progressBar.setVisibility(View.GONE);
+          errorTextView.setVisibility(View.VISIBLE);
+          errorTextView.setText(getResources().getString(R.string.error_failed_login));
+          Toast.makeText(this.getActivity(), loginResultResource.message, Toast.LENGTH_SHORT).show();
+          break;
+        case SUCCESS:
+          progressBar.setVisibility(View.GONE);
+          errorTextView.setVisibility(View.GONE);
+          // save the token and set the authInterceptor
+          tokenAuthInterceptor.setJwtToken(loginResultResource.data.token);
```

Comment:
Since you have applied MVVM so nicely, I wanted to point out that business / networking logic does not belong in the UI. It should be done in the ViewModel. Let the View care about only UI stuff and navigation